File: 0704-binary-search/0704-binary-search.java
class Solution {
    public int search(int[] nums, int target) {
        
        int lo=0, hi=nums.length-1;
        
        while(lo != hi) {
            int mid = (int) Math.floor((lo+hi)/2);
            // if (nums[mid] == target) {
            //     return mid;
            // }
            // else 
            if (nums[mid] < target) {
                lo = mid+1;
            }
            else {
                hi = mid;
            }
        }
        if (nums[lo] == target) {
            return lo;
        }
        else {
            return -1;
        }
        
    }
}

File: 0704-binary-search/0704-binary-search.py
class Solution:
    def search(self, nums: List[int], target: int) -> int:
        
        lo = 0
        hi = len(nums) - 1
        
        while (lo <= hi):
            mid = (lo+hi) // 2
            
            ey = nums[mid]
            
            if (target < ey):
                hi = mid-1
            elif (target > ey):
                lo = mid+1
            else:
                return mid
        
        return -1
        